evm/src/cross_table_lookup.rs

```diff
@@ -21,42 +23,127 @@ use crate::proof::{StarkProofWithPublicInputs, StarkProofWithPublicInputsTarget}
 use crate::stark::Stark;
 use crate::vars::{StarkEvaluationTargets, StarkEvaluationVars};
 
+/// Represent a linear combination of columns.
 #[derive(Clone)]
-pub struct TableWithColumns {
-    table: Table,
-    columns: Vec<usize>,
-    /// Vector of columns `[c_1,...,c_k]` used as a filter using the sum `c_1 + ... + c_k`.
-    /// An empty vector corresponds to no filter.
-    filter_columns: Vec<usize>,
+pub struct Column<F: Field> {
+    linear_combination: Vec<(usize, F)>,
+    constant: F,
 }
 
-impl TableWithColumns {
-    pub fn new(table: Table, columns: Vec<usize>, filter_columns: Vec<usize>) -> Self {
+impl<F: Field> Column<F> {
+    pub fn single(c: usize) -> Self {
+        Self {
+            linear_combination: vec![(c, F::ONE)],
+            constant: F::ZERO,
+        }
+    }
+
+    pub fn singles(cs: Vec<usize>) -> Vec<Self> {
+        cs.into_iter().map(Self::single).collect()
+    }
```

Comment:
I suggest making changing this signature to `singles<I: IntoIterator<Item = usize>>(cs: I) -> impl Iterator<Item = Self>`. It would be slightly more flexible (and saves you a `.collect()`).